src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -293,48 +292,57 @@ public DateTime(int year, int month, int day, int hour, int minute, int second,
             if (second != 60 || !SystemSupportsLeapSeconds)
             {
                 ulong ticks = calendar.ToDateTime(year, month, day, hour, minute, second, millisecond).UTicks;
-                _dateData = ticks | ((ulong)kind << KindShift);
+                _dateData = ticks | ((ulong)(uint)kind << KindShift);
             }
             else
             {
-                // if we have a leap second, then we adjust it to 59 so that DateTime will consider it the last in the specified minute.
-                this = new DateTime(year, month, day, hour, minute, 59, millisecond, calendar, kind);
-                ValidateLeapSecond();
+                _dateData = WithLeapSecond(calendar, year, month, day, hour, minute, millisecond, kind);
             }
         }
 
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static ulong WithLeapSecond(Calendar calendar, int year, int month, int day, int hour, int minute, int millisecond, DateTimeKind kind)
```

Comment:
What is the benefit of this manual outlining? Does the ctor get inlined automatically with this factored out but not with it  left in line?